url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/log/Log.java

```diff
@@ -0,0 +1,83 @@
+package net.hetimatan.util.log;
+
+import java.io.File;
+import java.io.IOException;
+
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.event.GlobalAccessProperty;
+
+
+public class Log {
+	public static final boolean ON = true;//false;//true;//false;//true;//false;//true;//alse;//false;//false; 
+	public static final long start = System.nanoTime();
+    public static final int  MODE_STDOUT = 1;
+    public static final int  MODE_FILE = 2;
+    public static final Log  log = new Log();
+    public static final int  mode = 1;//MODE_FILE;
+    public static final boolean isTime = false;
+    public static final int loglevel = 5;//8;// 5 v, 8 d
+    private CashKyoroFile mLog = null;
+  
+    public Log() {
+    	if(mode == MODE_FILE && ON) {
+    		File parent = (new File("dummy")).getParentFile();
+    		String path = GlobalAccessProperty.getInstance().get("my.home", parent.getAbsolutePath());
+    		File home = new File(path);
+            File file = new File(home, "master.log");
+            try {
+                mLog = new CashKyoroFile(file, 512, 2);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *mLog*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).